src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs

```diff
@@ -736,6 +739,106 @@ private static bool TryCompareTimeZoneInformationToRegistry(in TIME_ZONE_INFORMA
+                {
+                    return result;
+                }
+            }
+
+            result = GetFileMuiPath(filePath, cultureInfo);
+
+            lock (s_FileMuiPathCache)
+            {
+                s_FileMuiPathCache.TryAdd(cacheKey, result);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Helper function for getting the MUI path for a given resource and culture.
+        /// Searches the installed OS languages for either an exact matching culture, or one that has the same parent.
+        /// If not found, uses the preferred default OS UI language, to align with prior behavior.
+        /// </summary>
+        private static unsafe string GetFileMuiPath(string filePath, CultureInfo cultureInfo)
+        {
+            char* fileMuiPath = stackalloc char[Interop.Kernel32.MAX_PATH + 1];
+            char* language = stackalloc char[Interop.Kernel32.LOCALE_NAME_MAX_LENGTH + 1];
+            uint fileMuiPathLength = Interop.Kernel32.MAX_PATH;
+            uint languageLength = Interop.Kernel32.LOCALE_NAME_MAX_LENGTH;
+            ulong enumerator = 0;
+
+            while (true)
+            {
+                // Search all installed languages.  The enumerator is re-used between loop iterations.
+                bool succeeded = Interop.Kernel32.GetFileMUIPath(
+                    Interop.Kernel32.MUI_USE_INSTALLED_LANGUAGES,
+                    filePath, language, ref languageLength,
+                    fileMuiPath, ref fileMuiPathLength, ref enumerator);
+
+                if (!succeeded)
+                {
+                    // Recurse to search using the parent of the desired culture.
+                    if (cultureInfo.Parent.Name != string.Empty)
+                    {
+                        return GetFileMuiPath(filePath, cultureInfo.Parent);
+                    }
+
+                    // Final fallback, using the preferred installed UI language.
+                    enumerator = 0;
+                    succeeded = Interop.Kernel32.GetFileMUIPath(
+                        Interop.Kernel32.MUI_USER_PREFERRED_UI_LANGUAGES,
+                        filePath, language, ref languageLength,
+                        fileMuiPath, ref fileMuiPathLength, ref enumerator);
+
+                    if (succeeded)
+                    {
+                        fileMuiPath[Interop.Kernel32.MAX_PATH] = '\0';
+                        return new string(fileMuiPath);
+                    }
+
+                    // Shouldn't get here, as there's always at least one language installed.
+                    return string.Empty;
+                }
+
+                // Lookup succeeded.  Check for exact match to the desired culture.
+                language[Interop.Kernel32.LOCALE_NAME_MAX_LENGTH] = '\0';
+                var lang = new string(language);
+                if (string.Equals(lang, cultureInfo.Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
We shouldn't need to allocate the string just to do this comparison.  Presumably exact match is common and we'd want to delay the creation of lang until after this if block?  This can instead be:
```C#
ReadOnlySpan<char> lang = MemoryMarshal.CreateReadOnlySpanFromNullTerminated(language);
if (lang.Equals(cultureInfo.Name, StringComparison.OrdinalIgnoreCase)
```